pipeline/test/test_pipeline.py

```diff
@@ -74,6 +77,41 @@ def test_single_task(self):
         self.assertEqual(self.read(output_file), msg)
         self.rm(output_file)
 
+    def test_single_task_w_input(self):
+        msg = 'abc'
+        input_file = '/tmp/data/example1.txt'
+        output_file = '/tmp/test_single_task_w_input.txt'
+        self.write(input_file, 'abc')
+
+        p = Pipeline()
+        input = p.read_input(input_file)
+        t = p.new_task()
+        t.command(f'cat {input} > {t.ofile}')
+        p.write_output(t.ofile, output_file)
+        p.run()
+
+        self.assertEqual(self.read(output_file), msg)
+        self.rm(input_file, output_file)
```

Comment:
this should be in a finally block, probably? unless you purposely want to leave it when the assert fails